tls/s2n_server_new_session_ticket.c

```diff
@@ -125,6 +125,9 @@ S2N_RESULT s2n_tls13_server_nst_send(struct s2n_connection *conn, s2n_blocked_st
         return S2N_RESULT_OK;
     }
 
+    /* Flush any pending messages. */
+    RESULT_GUARD_POSIX(s2n_flush(conn, blocked));
+
```

Comment:
Why do we need to flush the NST message when we didn't need to before?

Reply:
I think it is safer to flush anytime we are about to write to the out stuffer. This is less of a problem if the out stuffer is growable (Will be in next rev).

Reply:
https://github.com/aws/s2n-tls/pull/3326#discussion_r886106842 Why is this safer? Is the concern that the conn->out stuffer can be full, and why was that not a concern when the stuffer could only fit one record?

Reply:
I think it safer to call flush before manipulating `conn->out` now that I've modified `s2n_record_writev()` to be able to fit more than one record of data into `conn->out`. 

Previously `conn->out` was always flushed after a record was created and had a check to make sure `conn->out` was empty. 

The post handshake logic can now be triggered in between flushes and to avoid a situation that `conn->out` cannot store any more data (Shouldn't happen) a defensive flush helps prevent that scenario.

Reply:
I don't see any tests for the case of post handshake messages being sent in the middle of application data records. Can we add those test cases?

Reply:
Yes will add some test cases for that

Reply:
I still don't see any test cases for post-handshake messages interrupting the buffered read-- am I missing them?